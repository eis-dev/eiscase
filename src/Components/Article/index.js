import * as React from "react";
import Link from "next/link";

export default class Article extends React.Component {
    constructor(props) {
        super(props);
    }

    contentHTML() {
        return this.props.article.content.split("\n").map((item) => <p>{item}</p>)
    };

    readMore() {
        return (
            <p className={"mb-0"}>
                <Link href={"/article/" + this.props.article.id}>Read more</Link>
            </p>
        )
    }

    render() {
        return (
            <div>
                <div className={!this.props.isDetail ? 'card mb-3' : ''}>
                    <div className={!this.props.isDetail ? 'card-body' : ''}>
                        <h5 className={"card-title"}>{this.props.article.title}</h5>
                        <p className={"card-text"}>
                            {this.props.isDetail ? this.contentHTML() : this.props.article.content.substr(0, 225) + "... "}
                        </p>
                        {!this.props.isDetail ? this.readMore() : ''}
                    </div>
                </div>
            </div>
        )
    }
}

export {Article};