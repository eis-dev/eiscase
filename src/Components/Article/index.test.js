import React from 'react';
import Article from './index';

import renderer from 'react-test-renderer';

const dummy = {
    title: "Test Title",
    content: "Test Content",
}

describe('Article', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Article article={dummy}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})