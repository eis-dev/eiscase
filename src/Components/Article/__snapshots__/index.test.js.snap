// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<div>
  <div
    className="card mb-3"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Test Title
      </h5>
      <p
        className="card-text"
      >
        Test Content... 
      </p>
      <p
        className="mb-0"
      >
        <a
          href="/article/undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Read more
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <div
    className="card mb-3"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Test Title
      </h5>
      <p
        className="card-text"
      >
        Test Content... 
      </p>
      <p
        className="mb-0"
      >
        <a
          href="/article/undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Read more
        </a>
      </p>
    </div>
  </div>
</div>
`;
